### PR TITLE
Dev/code cleaning and documentation

### PR DESCRIPTION
**Pull request by TheBlueHawk**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/firestore/ConcreteFirestoreInteractor.java [hunk 1]
```diff
@@ -36,67 +39,37 @@ public ConcreteFirestoreInteractor() {
...
-                        }})
-                    .exceptionally(e -> Collections.emptyMap());
-        } finally {
-            serverIdlingResource.decrement();
-        }
+    public CompletableFuture<Map<String, Map<String, Object>>> readCollection(CollectionReference collectionReference) {
+        Task<QuerySnapshot> collectionTask = collectionReference.get();
+        CompletableFuture<QuerySnapshot> completableFuture = taskToFuture(collectionTask);
+        return completableFuture
+                .thenApply(collection -> {
```
[the remark this answers is not in the document]
Comment (TheBlueHawk) [reply app/src/main/java/ch/epfl/sdp/firestore/ConcreteFirestoreInteractor.java:41]: Rule of 3 :)

File: app/src/main/java/ch/epfl/sdp/Map/PathsHandler.java [hunk 2]
```diff
@@ -165,19 +189,27 @@ private void mapStyle(Layer layer, Geometry geometry, String sourceId) {
...
-                .orderBy("Position.timestamp")
-                .get()
-                .addOnCompleteListener(task -> {
-                    if (task.isSuccessful()) {
-                        callback.onCallback(task.getResult().iterator());
+    private CompletableFuture<Map<String, Map<String, Object>>> initFirestorePathRetrieval() {
+        return FirestoreInteractor.taskToFuture(
+                collectionReference("History/USER_PATH_DEMO" + "/Positions")
+                        .orderBy("Position" + ".timestamp").get())
+                .thenApply(collection -> {
```
[the remark this answers is not in the document]
Comment (TheBlueHawk) [reply app/src/main/java/ch/epfl/sdp/Map/PathsHandler.java:196]: Rule of 3 :)

File: app/src/main/java/ch/epfl/sdp/firestore/ConcreteFirestoreInteractor.java [hunk 3]
```diff
@@ -16,11 +14,16 @@
...
+ * This class is the main and more generic implementation of the FirestoreInteractor, every other
+ * implementation should inherit from this one.
+ */
 public class ConcreteFirestoreInteractor extends FirestoreInteractor {
-    private final CountingIdlingResource serverIdlingResource;
 
+    /**
+     * Default constructor
+     */
     public ConcreteFirestoreInteractor() {
```
Comment (TheCrazyKing) [inline app/src/main/java/ch/epfl/sdp/firestore/ConcreteFirestoreInteractor.java]: Empty constructor ?

**Review by TheCrazyKing — APPROVED** [review]

File: app/src/main/java/ch/epfl/sdp/HistoryFirestoreInteractor.java [hunk 4]
```diff
@@ -24,7 +26,7 @@ private String historyPositionsPath(){
     }
 
 
-    public CompletableFuture<Void> write(Map<String, Object> content) {
+    public CompletableFuture<Void> writePositions(Map<String, Object> content) {
```
Comment (lucieho) [inline app/src/main/java/ch/epfl/sdp/HistoryFirestoreInteractor.java:29]: Good idea, it's more intuitive

File: app/src/main/java/ch/epfl/sdp/Tools.java [hunk 5]
```diff
@@ -1,25 +1,33 @@
...
-import java.util.concurrent.Executor;
+import com.google.firebase.firestore.GeoPoint;
 
-import static com.mapbox.mapboxsdk.Mapbox.getApplicationContext;
-
-public class BiometricUtils {
+/**
+ * This class includes some utilities used when dealing with biometric authentication
+ */
+public class Tools {
```
Comment (lucieho) [inline app/src/main/java/ch/epfl/sdp/Tools.java:20]: It's great that you moved the tools related to network status here! Makes more sense indeed

File: app/src/main/java/ch/epfl/sdp/firestore/FirestoreInteractor.java [hunk 6]
```diff
@@ -8,8 +8,60 @@
...
+    private static FirebaseFirestore firestore = FirebaseFirestore.getInstance();
+
+    /**
+     * Convert a task into a completableFuture
+     *
+     * @param task The task we want to convert to a CompletableFuture
+     * @param <T>  The type of future required
+     * @return a future behaving as the original task
+     */
+    public static <T> CompletableFuture<T> taskToFuture(Task<T> task) {
```
Comment (lucieho) [inline app/src/main/java/ch/epfl/sdp/firestore/FirestoreInteractor.java:26]: Pretty useful method!

**Review by lucieho — APPROVED** [review]
Nice job you've done here!

File: app/src/main/java/ch/epfl/sdp/firestore/ConcreteFirestoreInteractor.java (hunk 3 above)
Comment (TheBlueHawk) [reply app/src/main/java/ch/epfl/sdp/firestore/ConcreteFirestoreInteractor.java]: Removed

**Pull request merged**